.circleci/config.yml

```diff
@@ -253,6 +260,81 @@ jobs:
                 include_job_number_field: false
                 include_project_field: false
                 failure_message: ':red_circle: Scheduled tests failed on iOS device!'
+  android-build-react-native-bridge:
+    # Running on a Ubuntu machine to have more memory compared to the Docker +
+    # Node image that CircleCI offers (circleci/node)
+    #
+    # See: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3038#issuecomment-765334675
+    machine:
+      image: << pipeline.parameters.linux-machine-image >>
+    steps:
+      - checkout
+      - checkout-submodules
+      - npm-install
+      - run:
+          name: Build JS Bundle
+          command: |
+            npm run bundle:android
+          # This command may take a long time, so prevent CircleCI from killing
+          # the build if it doesn't receive output for more than its default
+          # ten minutes.
+          no_output_timeout: 30m
+      # Move the generated bundle to the location where we'll need it for the
+      # Gradle build, then put it in the CI cache. We'll get it from there in
+      # the publishing job, which runs on a Docker container optimized for
+      # Android work.
+      - run:
+          name: Ensure assets folder exists
+          command: mkdir -p gutenberg/packages/react-native-bridge/android/build/assets
+      - run:
+          name: Move bundle to assets folder
+          command: mv bundle/android/App.js gutenberg/packages/react-native-bridge/android/build/assets/index.android.bundle
+      - get-cache-id-for-react-native-bridge
+      - save_cache:
+          name: Cache React Native Android Bridge JS Bundle
+          key: android-js-bundle-{{ checksum "cache_id" }}
+          paths:
+            - gutenberg/packages/react-native-bridge/android/build/assets/index.android.bundle
```

Comment:
@mokagio I don't fully understand the reasons for manually building the JS bundle when we have the tools in place for Gradle to do it for us. We currently need to update how `react-native-bridge` is built so we can tell to build the JS bundle when we are building for Bintray, but aside from that, we shouldn't really have any issue with Gradle building the JS bundle for us.

I have a few concerns with the manual bundle generation:
* We'll need to maintain two different build structures and run the risk of one working while the other is not
* It'll not be trivial to locally reproduce a CI build as we'll need to check/follow the CI tasks and follow the commands (especially for developers who are less experienced with the CI config)
* We can't easily test the binary dependency. If we make Gradle build the bundle, we can simply switch a flag and make WPAndroid use `react-native-bridge` as a "compiled" dependency. (mostly replicating a binary build)

I am not entirely sure what the benefits of manually building the JS bundle are. I thought maybe it'd be caching, but we can cache the bundle in a Gradle build as well, which I believe we already do in WPAndroid, so that's probably not it. Please let me know if I am missing something.

Reply:
@oguzkocer thank you for raising this point.

Building the bundle as a separate step from the Bintray build & upload is due to memory issues I've experienced with CircleCI. That's the only, unfortunate, reason.

The issues I saw were stuff like

```
FAILURE: Build failed with an exception.

* What went wrong:
Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)
```

([example build](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/9963/workflows/c7a585ce-1486-4fd6-ae3a-dfa566bd8852/jobs/52701))

or 

```
Process 'command 'npx'' finished with non-zero exit value 1
```

([example build](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/9856/workflows/92767fa2-4e82-47d4-a733-d36128076d91/jobs/52203))

and timeouts when Gradle runs the `buildJSBundle` task

```
Too long with no output (exceeded 10m0s): context deadline exceeded
```

([example build](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/9966/workflows/d8099944-d8f5-4a43-a40a-22b949c4c0b1/jobs/52709))

Having said all that... I learned a few things working on this and I took your feedback at heart. I pushed a [new branch](https://github.com/wordpress-mobile/gutenberg-mobile/tree/mokagio/publish-to-bintray-from-circle-gradle-only) that only runs the Gradle task and configured it to wait for no output for 30 minutes instead of the default 10 and uses [a version of Gutenberg that results in always building the bundle](https://github.com/mokagio/gutenberg/commit/e3dace).

The first build [succeeded](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/10302/workflows/07fc7917-1c0f-4290-97e8-884a9fc8c140/jobs/54469) the second [failed](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/10304/workflows/1ff19aaa-4496-4a4d-9afb-590d11905bf0/jobs/54476) with "Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)".

I'd love to get this to work in one step because I agree with your points but, so far, I haven't been able to.

Reply:
An idea that popped to mind but that I haven't had a chance to explore would be to try and use `machine: linux` instead of the Docker container. It worked to solve the memory issues with the `npm` task in isolation (as per previous discussion in this PR) and since that seems to be the issue when Gradle calls it, too, it might be a valuable option.

Reply:
I see.. In that case, would you mind if I also give it a shot to see if I can get it to work? I don't have much time right now, but if this is not blocking you, I can definitely look into it after the composite build for the react native demo app is working.

Reply:
I SSHed into the Linux machine in CircleCI and hacked a working Android setup. Once I got all the pieces in place, it built on the first go 😱 

![image](https://user-images.githubusercontent.com/1218433/106705285-c6412e80-6641-11eb-90cd-becfa2822619.png)

The time it took is atrocious, but in line with what we're already getting for the standalone Node build step.